packages/zoe/src/types.js

```diff
@@ -349,17 +347,86 @@
  * @property {Issuer} issuer
  * @property {AmountMath} amountMath
  *
- * @typedef {Object} InstallationRecord
- * @property {InstallationHandle} handle - opaque identifier for the installation
- * @property {SourceBundle} bundle - contract code
- *
- * @typedef {Object} OfferStatus
- * @property {OfferHandle[]} active
- * @property {OfferHandle[]} inactive
- *
- * @typedef {(offerHandle: OfferHandle) => void} OfferHook
+ * @typedef {Record<Keyword,Amount>} Allocation
+ * @typedef {Record<Keyword,AmountMath>} AmountMathKeywordRecord
+ */
+
+/**
+ * @typedef {Object} ZCFSeat
+ * @property {() => void} exit
+ * @property {(msg?: string) => void} kickOut
+ * @property {() => Notifier<Allocation>} getNotifier
+ * @property {() => boolean} hasExited
+ * @property {() => ProposalRecord} getProposal
+ * @property {(keyword: Keyword, brand: Brand) => Amount} getAmountAllocated
+ * The brand is used for filling in an empty amount if the `keyword`
+ * is not present in the allocation
+ * @property {() => Allocation} getCurrentAllocation
+ * @property {(newAllocation: Allocation) => Boolean} isOfferSafe
+ * @property {(newAllocation: Allocation) => SeatStaging} stage
+ */
+
+/**
+ * @typedef {Object} SeatStaging
+ * @property {() => ZCFSeat} getSeat
+ * @property {() => Allocation} getStagedAllocation
+ */
+
+/**
+ * @typedef {{ zcfSeat: ZCFSeat, userSeat: ERef<UserSeat>}} ZcfSeatKit
+ */
+
+/**
+ * @callback OfferHandler
+ * @param {ZCFSeat} seat
+ * @returns any
+ */
+
+/**
+ * @typedef {Object} ContractStartFnResult
+ * @property {Object=} creatorFacet
+ * @property {Promise<Invitation>=} creatorInvitation
+ * @property {Object=} publicFacet
+ */
+
+/**
+ * @callback ContractStartFn
+ * @param {ContractFacet} zcf
+ * @param {Object} terms
```

Comment:
This parameter goes away in the next reform?

Reply:
Yes